### PR TITLE
issue-6: bugfixes, refactor, performance improvements

### PR DESCRIPTION
**Pull request by oliverfoster**
https://github.com/adaptlearning/jquery.onscreen/issues/6

### CODE REVIEW

**Comment by chris-steele** [comment]
+1

File: jquery.onscreen.js [hunk 1]
```diff
@@ -1,291 +1,406 @@
...
-			var items = inViewHandlers;
-			for (var i = 0; i < items.length; i++) {
-				var item = items[i];
-				triggerInview(item);
-			}
-		}
+		repeat: function() {
 
-	}
+			var intervalAttached = (loop.timeoutHandle !== null);
```
Comment (tomgreenfield) [inline jquery.onscreen.js]: Has ```intervalAttached``` just been pasted from line 223? Doesn't seem to be used here.

File: jquery.onscreen.js [hunk 2]
```diff
@@ -1,291 +1,406 @@
...
-		wScrollLeft = $window.scrollLeft();
 	}
 
-	$(window).on("scroll", startLoop);
-	$(window).on("mousedown touchstart keydown", startLoop);
-	$(window).on("resize", windowResize);
+	//attach event handlers
+	$(window).on("mousedown touchstart keydown", loop.start);
+	$(window).on("scroll", loop.start);
+	$(window).on("resize", wndw.resize);
```
Comment (tomgreenfield) [inline jquery.onscreen.js]: Minor again – can we only query the selector once here?
```js
$(window).on({
    "mousedown touchstart keydown": loop.start,
    "scroll": loop.start,
    etc...
});
```

Comment (oliverfoster) [inline jquery.onscreen.js]: i always forget about the object syntax. let me get @lc-thomasberger's caching in and i'll change those lines..

**Pull request merged**